client/my-sites/stats/features/modules/stats-locations/stats-locations.tsx

```diff
@@ -124,32 +163,95 @@ const StatsLocations: React.FC< StatsModuleLocationsProps > = ( {
 
 	// Main location data query
 	const {
-		data: locationsViewsData = [],
+		data: locationsViewsData,
```

Comment:
This was cuasing the variables to be recreated so moved this condition in memo below